pyro/distributions/random_primitive.py

```diff
@@ -35,9 +35,9 @@ def event_dim(self, *args, **kwargs):
         kwargs.pop('sample_shape', None)
         return self.dist_class(*args, **kwargs).event_dim()
 
-    def shape(self, x=None, *args, **kwargs):
+    def shape(self, *args, **kwargs):
         kwargs.pop('sample_shape', None)
-        return self.dist_class(*args, **kwargs).shape(x)
+        return self.dist_class(*args, **kwargs).shape()
```

Comment:
Now is this the correct behavior, or should `.shape()` also include the `sample_shape` that was popped on the previous line? Again I'm not sure what our semantics is.

Reply:
I was considering removing even the `shape` method later, and replacing with `batch_shape + event_shape` instead. But I think it will be useful to retain it and prepend `sample_shape` before. Will clean this up.

Reply:
Yeah I'm fine removing the ambiguously named `.shape()` method.